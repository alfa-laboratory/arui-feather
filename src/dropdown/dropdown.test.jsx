/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { mount } from 'enzyme';

import Dropdown from './dropdown';

describe('dropdown', () => {
    it('should render without problems', () => {
        let dropdown = mount(
            <Dropdown popupContent='Popup-example'>Switcher-example</Dropdown>
        );

        expect(dropdown).toMatchSnapshot();
        expect(dropdown.find('.popup').text()).toBe('Popup-example');
        expect(dropdown.find('a.dropdown__switcher').text()).toBe('Switcher-example');
    });

    it('should render link switcher by default', () => {
        let dropdown = mount(
            <Dropdown />
        );

        expect(dropdown.find('a.dropdown__switcher').props().className).toContain('link');
    });

    it('should render button switcher with property switcherType=`button`', () => {
        let dropdown = mount(
            <Dropdown switcherType='button' />
        );

        expect(dropdown.find('button.dropdown__switcher').props().className).toContain('button');
    });

    it('should call `onSwitcherMouseEnter` callback after switcher was hovered', () => {
        let onSwitcherMouseEnter = jest.fn();
        let dropdown = mount(
            <Dropdown onSwitcherMouseEnter={ onSwitcherMouseEnter } switcherType='button' />
        );
        let switcherNode = dropdown.find('button.dropdown__switcher');

        switcherNode.simulate('mouseEnter');

        expect(onSwitcherMouseEnter).toHaveBeenCalled();
    });

    it('should call `onSwitcherMouseLeave` callback after switcher was unhovered', () => {
        let onSwitcherMouseLeave = jest.fn();
        let dropdown = mount(
            <Dropdown onSwitcherMouseLeave={ onSwitcherMouseLeave } switcherType='button' />
        );
        let switcherNode = dropdown.find('button.dropdown__switcher');

        switcherNode.simulate('mouseLeave');

        expect(onSwitcherMouseLeave).toHaveBeenCalled();
    });

    it('should call `onSwitcherClick` callback after switcher was clicked', () => {
        let onSwitcherClick = jest.fn();
        let dropdown = mount(
            <Dropdown onSwitcherClick={ onSwitcherClick } switcherType='button' />
        );
        let switcherNode = dropdown.find('button.dropdown__switcher');

        switcherNode.simulate('click');

        expect(onSwitcherClick).toHaveBeenCalled();
    });

    it('should not call `onSwitcherClick` callback after switcher was clicked with disabled=true', () => {
        let onSwitcherClick = jest.fn();
        let dropdown = mount(
            <Dropdown onSwitcherClick={ onSwitcherClick } switcherType='button' disabled={ true } />
        );
        let switcherNode = dropdown.find('button.dropdown__switcher');

        switcherNode.simulate('click');

        expect(onSwitcherClick).not.toHaveBeenCalled();
    });

    it('should call `onPopupMouseEnter` callback after popup was hovered', () => {
        let onPopupMouseEnter = jest.fn();
        let dropdown = mount(
            <Dropdown onPopupMouseEnter={ onPopupMouseEnter } opened={ true } />
        );
        let popupNode = dropdown.find('.popup');

        popupNode.simulate('mouseEnter');

        expect(onPopupMouseEnter).toHaveBeenCalled();
    });

    it('should call `onPopupMouseLeave` callback after popup was unhovered', () => {
        let onPopupMouseLeave = jest.fn();
        let dropdown = mount(
            <Dropdown onPopupMouseLeave={ onPopupMouseLeave } opened={ true } />
        );
        let popupNode = dropdown.find('.popup');

        popupNode.simulate('mouseLeave');

        expect(onPopupMouseLeave).toHaveBeenCalled();
    });

    it('should set class checked to switcher button then togglable=`check` and popup open', () => {
        let dropdown = mount(
            <Dropdown togglable='check' opened={ true } switcherType='button' />
        );
        let switcherNode = dropdown.find('button.dropdown__switcher');

        expect(switcherNode.props().className).toContain('button_checked');
    });

    it('should toggle popup visibility by click on switcher', () => {
        let dropdown = mount(
            <Dropdown />
        );
        let switcherNode = dropdown.find('a.dropdown__switcher');

        switcherNode.simulate('click');
        expect(dropdown.find('.popup').props().className).toContain('popup_visible');

        switcherNode.simulate('click');
        expect(dropdown.find('.popup').props().className).not.toContain('popup_visible');
    });
});
