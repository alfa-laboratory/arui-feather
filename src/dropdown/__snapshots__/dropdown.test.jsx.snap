// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown should render without problems 1`] = `
<Dropdown
  disabled={false}
  popupContent="Popup-example"
  popupProps={
    Object {
      "target": "anchor",
    }
  }
  size="m"
  switcherText="Switcher"
  switcherType="link"
>
  <div
    className="dropdown dropdown_theme_alfa-on-white"
  >
    <Link
      checked={false}
      className="dropdown__switcher"
      disabled={false}
      iconPosition="left"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      pseudo={true}
      size="m"
      tabIndex={0}
      text="Switcher-example"
      url="#"
    >
      <a
        className="link link_pseudo link_size_m link_theme_alfa-on-white dropdown__switcher"
        href="#"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        tabIndex={0}
      >
        <span
          className="link__text"
          key="text"
        >
          Switcher-example
        </span>
      </a>
    </Link>
    <Popup
      className="dropdown__popup"
      fitContaiterOffset={0}
      mainOffset={6}
      onClickOutside={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      padded={true}
      secondaryOffset={0}
      size="m"
      target="anchor"
      visible={false}
    >
      <div
        className="popup popup_direction_bottom-left popup_target_anchor popup_size_m popup_padded popup_theme_alfa-on-white dropdown__popup"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "bottom": undefined,
            "height": "auto",
            "left": 0,
            "maxHeight": "none",
            "maxWidth": "none",
            "minWidth": 0,
            "right": undefined,
            "top": 6,
          }
        }
      >
        <div
          className="popup__container"
        >
          <div
            className="popup__inner"
            onScroll={[Function]}
          >
            <div
              className="popup__content"
            >
              Popup-example
              <ResizeSensor
                onResize={[Function]}
              >
                <iframe
                  style={
                    Object {
                      "background": "transparent",
                      "border": "none",
                      "height": "100%",
                      "left": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                      "zIndex": -1,
                    }
                  }
                  tabIndex="-1"
                />
              </ResizeSensor>
            </div>
          </div>
        </div>
      </div>
    </Popup>
  </div>
</Dropdown>
`;
